### PR TITLE
grpc: add pagination support to GetTransactions

### LINKED ISSUE #768: WalletService.GetTransactions should stream results as they are available
Currently it does write to a grpc stream but only after the entire result set is ready.  This is unacceptable from a performance and memory usage standpoint.

### PR DESCRIPTION
**Pull request by matheusd**
Fix #768. 

Still a POC at this point.

This changes the behaviour of the `(*WalletServer) GetTransaction` from returning the full set of transactions to calling a function for each result (following the same convention as `txStore.RangeTransactions()`).

This also implements a new request parameter (`maximum_transaction_count`) to allow a client to ask for transactions inside a given block window up to a given maximum, to allow time-based pagination (by repeatedly asking for increasing start..end intervals)

Still missing:

- [x] Documenting new function signatures
- [x] Document new api parameter

### CODE REVIEW

File: wallet/wallet.go [hunk 1]
```diff
@@ -2343,29 +2344,28 @@ func (w *Wallet) GetTransactions(ctx context.Context, startBlock, endBlock *Bloc
...
 					Height:       details[0].Block.Height,
 					Timestamp:    details[0].Block.Time.Unix(),
 					Transactions: txs,
-				})
+				}
 			} else {
-				res.UnminedTransactions = txs
+				block = &Block{
+					Height:       -1,
+					Transactions: txs,
```
Comment (matheusd) [inline wallet/wallet.go]: I'm using a block height of -1 to signal `f()` that the transactions are unmined. I did it like this to reduce the number of arguments to `f`, but can rework into separate `f(minedBlock, unimedTx)` if needed

File: rpc/rpcserver/server.go [hunk 2]
```diff
@@ -904,31 +904,43 @@ func (s *walletServer) GetTransactions(req *pb.GetTransactionsRequest,
 				"minimum number of recent transactions may not be negative")
 		}
 	}
-
 	_ = minRecentTxs
```
Comment (matheusd) [inline rpc/rpcserver/server.go]: I think I know where you were going with the `minimum_recent_transactions` but I think a better pagination strategy is to call with a begin/end window and a max count of transactions (cause then I can get transactions both in increasing/decreasing intervals by adjusting the begin/end indices accordingly).

But if there is another plan for this, let me know so I can rework it.

**Comment by alexlyp** [comment]
Looks like you need a small version bump resolution

File: rpc/api.proto [hunk 3]
```diff
@@ -279,7 +279,8 @@ message GetTransactionsRequest {
 	// TODO: remove until spec adds it back in some way.
 	int32 minimum_recent_transactions = 5;
 
-	// TODO: limit max number of txs?
+	// Include at most this many transactions in the reply
+	int32 maximum_transaction_count = 6;
```
Comment (jrick) [inline rpc/api.proto]: Maybe name this target_transaction_count since it isn't the maximum?

**Comment by jrick** [comment]
Please squash this with a complete commit message describing the change.

**Pull request merged**